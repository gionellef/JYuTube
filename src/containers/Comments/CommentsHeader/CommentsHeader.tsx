import * as React from 'react';

import { Button, Icon } from 'semantic-ui-react';

import './CommentsHeader.scss';

interface Props {
  numberOfComments: number;
}

const CommentsHeader : React.FC<Props> = (props) => {
  return (
    <div className='comments-header'>
      <h4>{ props.numberOfComments } Comments</h4>
      <Button basic compact icon labelPosition='left'>
        <Icon name='align left' />
        Sort by
      </Button>
    </div>
  );
}

export default CommentsHeader;