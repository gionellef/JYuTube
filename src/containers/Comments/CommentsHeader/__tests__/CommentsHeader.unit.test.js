import React from 'react';

import { shallow } from 'enzyme';

import { CommentsHeader } from '../CommentsHeader';

describe('CommentsHeader', () => {
  test('CommentsHeader renders with props.numberOfComments = null', () => {
    const wrapper = shallow(
      <CommentsHeader />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('CommentsHeader renders with props.numberOfComments = 0', () => {
    const wrapper = shallow(
      <CommentsHeader numberOfComments={634} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});