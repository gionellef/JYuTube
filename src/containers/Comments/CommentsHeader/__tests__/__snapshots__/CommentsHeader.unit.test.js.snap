// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsHeader CommentsHeader renders with props.numberOfComments = 0 1`] = `
<div
  className="comments-header"
>
  <h4>
    634
     Comments
  </h4>
  <Button
    as="button"
    basic={true}
    compact={true}
    icon={true}
    labelPosition="left"
  >
    <Icon
      as="i"
      name="align left"
    />
    Sort by
  </Button>
</div>
`;

exports[`CommentsHeader CommentsHeader renders with props.numberOfComments = null 1`] = `
<div
  className="comments-header"
>
  <h4>
     Comments
  </h4>
  <Button
    as="button"
    basic={true}
    compact={true}
    icon={true}
    labelPosition="left"
  >
    <Icon
      as="i"
      name="align left"
    />
    Sort by
  </Button>
</div>
`;
